osquery/core/sqlite_util.cpp

```diff
@@ -63,4 +63,56 @@ int query_data_callback(void* argument,
   (*qData).push_back(r);
   return 0;
 }
+
+Status getQueryColumns(const std::string& q, tables::TableColumns& columns) {
+  sqlite3* db = createDB();
+  Status status = getQueryColumns(q, columns, db);
+  sqlite3_close(db);
+  return status;
+}
+
+Status getQueryColumns(const std::string& q,
+                       tables::TableColumns& columns,
+                       sqlite3* db) {
+  int rc;
+
+  // Will automatically handle calling sqlite3_finalize on the prepared stmt
+  // (Note that sqlite3_finalize is explicitly a nop for nullptr)
+  std::unique_ptr<sqlite3_stmt, decltype(sqlite3_finalize)*>
+    stmt_managed(nullptr, sqlite3_finalize);
+  sqlite3_stmt* stmt = stmt_managed.get();
+
+  // Turn the query into a prepared statement
+  rc = sqlite3_prepare_v2(db, q.c_str(), q.length() + 1, &stmt, nullptr);
+  if (rc != SQLITE_OK) {
+    return Status(1, sqlite3_errmsg(db));
+  }
+
+  // Get column count
+  int num_columns = sqlite3_column_count(stmt);
+  std::vector<std::pair<std::string, std::string> > results;
```

Comment:
Could you remove the intermediate `tables::TableColumns` data structure and accumulate into `columns` directly?